src/client/app/router.ts

```diff
@@ -226,7 +226,9 @@ export function createRouter(
       loadPage(
         normalizeHref(location.href),
         (e.state && e.state.scrollPosition) || 0
-      )
+      ).then(() => {
+        router.onAfterRouteChanged?.(location.href)
```

Comment:
Use await expression instead
```suggestion
      await router.onAfterRouteChanged?.(location.href)
```

Reply:
Change:
```diff
-    window.addEventListener('popstate', (e) => {
-      loadPage(
+    window.addEventListener('popstate', async (e) => {
+      await loadPage(
        normalizeHref(location.href),
        (e.state && e.state.scrollPosition) || 0
-      ).then(() => {
-        router.onAfterRouteChanged?.(location.href)
       )
+      router.onAfterRouteChanged?.(location.href)
    })
```